### PR TITLE
Fix predicates for SELinux and ownership management

### PR DESCRIPTION
**Pull request by ghost**
For SELinux and ownership management we should check if:
- The volume supports it and
- The volume is not read-only or
- The volume is read-only but it is "managed"

This changes the checks to match the above.

@pmorie WDYT ?

### CODE REVIEW

**Comment by pmorie** [comment]
This patch is not correct and won't give the correct behavior.  I talked to @swagiaal about this on Friday and I will open a PR clarifying the existing code, but the behavior in master is correct as far as I know.

**Comment by swagiaal** [comment]
> I will open a PR clarifying the existing code

Sound good.

> the behavior in master is correct as far as I know.

Not quite because we still have the failures in the EmptyDir and the Security Context e2e's, but I have update the PR in light of our discussion on Friday. Should be correct now. PTAL.

@pmorie

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1086,7 +1086,7 @@ func makeMounts(pod *api.Pod, podDir string, container *api.Container, podVolume
 		// If the volume supports SELinux and it has not been
 		// relabeled already and it is not a read-only volume,
 		// relabel it and mark it as labeled
-		if vol.Builder.GetAttributes().SupportsSELinux && !vol.SELinuxLabeled && !vol.Builder.GetAttributes().Managed {
+		if vol.Builder.GetAttributes().SupportsSELinux && !vol.SELinuxLabeled && vol.Builder.GetAttributes().Managed {
```
Comment (pmorie) [inline pkg/kubelet/kubelet.go]: Let's put `Managed` first in the list.

**Comment by pmorie** [comment]
@swagiaal I confirmed the breakage in this test locally.  I don't understand yet why the secrets E2E didn't catch this.

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (pmorie) [inline pkg/kubelet/kubelet.go]: @swagiaal

**Comment by swagiaal** [comment]
@pmorie comment addressed

**Comment by swagiaal** [comment]
@k8s-bot test this

**Comment by swagiaal** [comment]
@k8s-bot test this.

**Comment by swagiaal** [comment]
@k8s-bot unit test this

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**